modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -42,6 +42,11 @@
 #ifndef OPENCV_DNN_DNN_HPP
 #define OPENCV_DNN_DNN_HPP
 
+#include <opencv2/dnn/csl/cublas.hpp>
+#include <opencv2/dnn/csl/cudnn.hpp>
+#include <opencv2/dnn/csl/stream.hpp>
+#include <opencv2/dnn/csl/workspace.hpp>
```

Comment:
You need to hide these headers from public header. I think that `initCUDA` params should be wrapper in the `BackendWrapper`.